astrapy/db.py

```diff
@@ -482,23 +486,22 @@ def __init__(
             )
             namespace = DEFAULT_KEYSPACE_NAME
 
-        # Store the initial parameters
+        # Store the API token
         self.token = token
-        (
-            self.database_id,
-            self.database_region,
-            self.database_domain,
-        ) = parse_endpoint_url(api_endpoint)
 
-        # Set the Base URL for the API calls
-        self.base_url = (
-            f"https://{self.database_id}-{self.database_region}.{self.database_domain}"
-        )
-        self.base_path = f"{DEFAULT_BASE_PATH}/{namespace}"
+        # Set the API version and path from the call
+        self.api_version = api_version or DEFAULT_JSON_API_VERSION
+        self.api_path = (api_path or DEFAULT_JSON_API_PATH).rstrip("/")
 
-        # Set the namespace parameter
+        # Set the namespace
         self.namespace = namespace
 
+        # Set the Base URL for the API calls
+        self.base_url = api_endpoint.rstrip("/")
+        self.base_path = posixpath.join(
```

Comment:
I have trouble with this line. `base_path` is not supposed to contain the domain, right? I.e. it should start at "/api/json/v1 ...." to my understanding. Yet there is self.base_url as the first parameter of posixpath.join.

I think the reason this now works is the wrong reason. Look:
```
>>> a=AstraDB("token", "https://aaa-bbb.datastax.com")
>>> a.base_path
'/api/json/v1/default_keyspace'
>>> a.base_url
'https://aaa-bbb.datastax.com'


>>> b=AstraDB("token", "https://aaa-bbb.datastax.com", api_path="trolling/you", api_version="vLOL")
>>> b.base_path
'https://aaa-bbb.datastax.com/trolling/you/vLOL/default_keyspace'
>>> b.base_url
'https://aaa-bbb.datastax.com'
```

See? `b.base_path` mistakenly contains the whole thing from https onward. `posixpath.join` forgets the parameters before a given one, if that one starts with `/` (it's a unix file system tool after all for what I can see).

So, `posixpath.join('a', 'b', 'c')` yelds `a/b/c`, but (and this is the brittle logic that underpins the above `posixpath.joins('a', 'b', '/c', 'd')` yields just `/c/d`.

I would be more fussy in stripping slashes left and right to each string, then reinstating manually the needed ones and explicitly not using self.base_url in constructing self.base_path - if I get this part right, that is.